src/overlay/SurveyManager.cpp

```diff
@@ -245,14 +632,57 @@ SurveyManager::makeSurveyRequest(NodeID const& nodeToSurvey) const
 void
 SurveyManager::sendTopologyRequest(NodeID const& nodeToSurvey)
 {
+    if (!mRunningSurveyReportingPhaseType)
+    {
+        throw std::runtime_error("Tried to send survey request when no survey "
```

Comment:
This function can be invoked via `topOffRequests` outside of an HTTP command, so I turned this exception into an error log + early return in c38e915095bde08ff9444d705d2ea2dd676dd63e. That being said, you're right that the condition that would trigger this exception is not possible. I changed to a log instead of removing entirely because I figured it's a good sanity check to log this as an error if some other code changes in the future accidentally invoke this function inappropriately.

I also checked all of the other exceptions I added and those are only reachable via HTTP commands and so should be safe.